core/src/main/java/org/apache/gravitino/connector/WildcardPropertiesMeta.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.connector;
+
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * WildcardPropertiesMeta is a utility class to validate wildcard properties in the properties
+ * metadata. <br>
+ * <br>
+ * WildcardPropertiesMeta interface defines: <br>
+ * FirstNode.SecondNode.WildcardNode = "" <br>
+ * FirstNode.SecondNode.*.property-key1 = "" <br>
+ * FirstNode.SecondNode.*.property-key2 = "" <br>
```

Comment:
Do wildcards have to be represented at the third node? It seems not general.

Reply:
I refactored the codes.